src/test/java/nova/core/event/EventBusTest.java

```diff
@@ -91,4 +95,86 @@ public void testRemovalByObject() {
 
 		assertThat(event.toString()).isEqualTo("A");
 	}
+
+	@Test
+	public void testNamedPriority1() {
+		EventBus<TestEvent> bus = new EventBus<>();
+		List<Character> result = new ArrayList<>();
+
+		bus.on().withName("A").before("B").bind(evt -> result.add('A'));
+		bus.on().withName("B").before("C").bind(evt -> result.add('B'));
+		bus.on().withName("C").bind(evt -> result.add('C'));
+
+		bus.publish(new TestEvent());
+
+		assertThat(result).hasSize(3);
+		assertThat(result).isSorted();
+	}
+
+	@Test
+	public void testNamedPriority2() {
+		EventBus<TestEvent> bus = new EventBus<>();
+		List<Character> result = new ArrayList<>();
+
+		bus.on().withName("C").after("B").bind(evt -> result.add('C'));
+		bus.on().withName("B").after("A").bind(evt -> result.add('B'));
+		bus.on().withName("A").bind(evt -> result.add('A'));
+
+		bus.publish(new TestEvent());
+
+		assertThat(result).hasSize(3);
+		assertThat(result).isSorted();
+	}
+
+	@Test
+	public void testNamedPriority3() {
```

Comment:
Do we need both types of test, I would combine this and the first test using only a string builder.